Upload request stats before inserting elements

amaltaro opened the pull request:
It should fix #5969 , though it was not tested yet.
I basically moved the block where stats are updated to couch so that if it fails, the request will have to wait until the next cycle.

@ticoann I also removed the block updating to wmstats, as we discussed. If you can take a quick look, that would be great. Just don't merge yet, it needs to be tested (though reproducing the failure will be very hard...)

amaltaro committed:
Upload request stats before inserting elements

add logdb logging; removed wmstats update

fix typo

pylint warnings cleaning

fixed python idioms; added file logging

remove comment

catch expection later

ericvaandering commented on src/python/WMCore/WorkQueue/WorkQueue.py:
```diff
@@ -1022,6 +1022,26 @@ def processInboundWork(self, inbound_work = None, throw = False, continuous = Fa
+                        except Exception as ex:
+                            msg = 'Error publishing stats %s to Request Mgr for %s: %s' % (totalStats,
+                                                                                           inbound['RequestName'],
+                                                                                           str(ex)))
```
Jenkins is not happy with this. I think you have an extra ')'

amaltaro replied:
Indeed! Good job! :)

amaltaro commented:
retest this please

ticoann commented:
@amaltaro, patch looks good to me. Should I merge this?

amaltaro commented:
Not yet please :( 
Normal tests were ok, but when I forced an exception in the try block, WQ elements were not created and they are not created even in the coming cycles... I'm still debugging it

amaltaro commented:
Thanks for the help Seangchan! Just updated the patch and seems to be working fine now.
The update call is already inside a try block, that's why I removed the inner one. Moreover, exceptions there will get propagated to queueNewRequests() which creates an error log entry in couch.

ticoann commented:
retest this please

amaltaro commented:
retest this please

amaltaro commented:
ok, I don't know what's going on... unittests are unstable, there were 23 failures in the last 2 runs and 170 in the last one. I don't find any info that tells me this PR is breaking tests.

ericvaandering commented:
I tried something yesterday to speed up the builds and rerun the parts that fail (more robust). I made a small typo though. Let's try again

ericvaandering commented:
retest this please

amaltaro commented:
Yaya, tests succeeded. Seangchan, feel free to merge :)
